### ISSUE TITLE
Release UltraSam model and US-43d dataset on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [CAMMA-public/UltraSam#18]
Hello Adrien Meyer and team 🤗

Niels here from the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2411.16222.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

It'd be great to make the UltraSam model checkpoints and the US-43d dataset compilation available on the 🤗 hub, to improve their discoverability/visibility.
We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

I see you've released the UltraSam model checkpoint at `https://s3.unistra.fr/camma_public/github/ultrasam/UltraSam.pth`. It would be awesome to host this directly on the Hugging Face Hub.

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset

Your US-43d dataset collection, comprising 43 datasets with over 280,000 images and segmentation masks, is a significant contribution. It would be awesome to make this unified dataset available on 🤗 , so that people can easily access and use it via:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
We could potentially help create a loading script for US-43d that can fetch the data components and apply the necessary preprocessing, making it readily available for the community.

See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by ben-jy** [comment]
Indeed, that would be great !
